packages/eslint-plugin/src/rules/restrict-plus-operands.ts

```diff
@@ -1,20 +1,21 @@
 import type { TSESTree } from '@typescript-eslint/utils';
+import * as tsutils from 'tsutils';
 import * as ts from 'typescript';
 
 import * as util from '../util';
 
 type Options = [
   {
-    checkCompoundAssignments?: boolean;
     allowAny?: boolean;
+    allowBoolean?: boolean;
+    allowNullish?: boolean;
+    allowNumberAndString?: boolean;
+    allowRegExp?: boolean;
```

Comment:
I don't think this is a good option to add - what are the usecases for summing a regex and another value?
The only real case I could think of would be to do something like `/a/ + 'u'` - but ultimately that case is pretty flaky given that you can accomplish the same thing in a safer way with `new RegExp(/a/, 'u')`.

I don't think that this option really makes sense in the context of the rule.